pkg/collector/corechecks/oracle-dbm/custom_queries.go

```diff
@@ -0,0 +1,153 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build oracle
+
+package oracle
+
+import (
+	"fmt"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	godror "github.com/godror/godror"
+	"reflect"
+	"strconv"
+)
+
+type Method func(string, float64, string, []string)
+
+type metricRow struct {
+	name   string
+	value  float64
+	method Method
+	tags   []string
+}
+
+func (c *Check) CustomQueries() error {
+	if c.dbCustomQueries == nil {
+		db, err := c.Connect()
```

Comment:
I don't fully understand why we're opening a new connection here. Can you explain?

Reply:
Custom queries is the only feature that switches to PDBs (all other queries are running against the root container). Switching to PDB and subsequent query execution isn't atomic, so there's no guarantee the both operations would get the same connection from the pool.

Reply:
Makes sense! Would appreciate a comment indicating this 🙏